### PR TITLE
Update to Parent POM 1.40 with configured FindBugs

### PR DESCRIPTION
**Pull request by oleg-nenashev**
Downstream of https://github.com/jenkinsci/pom/pull/18. Also picks Maven FindBugs Plugin 3.0.5

### Proposed changelog entries

* N/A

### Desired reviewers

@reviewbybees @dwnusbaum

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -492,28 +495,7 @@ THE SOFTWARE.
...
-            <execution>
-              <id>findbugs</id>
-              <goals>
-                <goal>check</goal>
-              </goals>
-              <phase>verify</phase>
-            </execution>
-          </executions>
-        </plugin>
+
```
Comment (dwnusbaum) [inline pom.xml]: nit: delete empty line

File: pom.xml [hunk 2]
```diff
@@ -92,8 +92,7 @@ THE SOFTWARE.
     <matrix-project.version>1.4.1</matrix-project.version>
     <sorcerer.version>0.11</sorcerer.version>
     <animal.sniffer.skip>${skipTests}</animal.sniffer.skip>
-    <findbugs-maven-plugin.version>3.0.4</findbugs-maven-plugin.version>
-    <findbugs.failOnError>true</findbugs.failOnError>
+
```
Comment (dwnusbaum) [inline pom.xml]: nit: delete empty line

File: pom.xml [hunk 3]
```diff
@@ -108,6 +107,10 @@ THE SOFTWARE.
     <remoting.version>3.17</remoting.version>
     <remoting.minimum.supported.version>2.60</remoting.minimum.supported.version>
 
+    <findbugs.failOnError>true</findbugs.failOnError>
```
Comment (dwnusbaum) [inline pom.xml]: No strong opinion, but this is the default, so it could be deleted, right?

**Review by dwnusbaum — APPROVED** [review]
Approach looks good to me, just have some nits.

**Review by jglick — APPROVED** [review]

**Review by dwnusbaum — APPROVED** [review]

**Pull request merged**